src/mesh/mesh.cpp

```diff
@@ -1149,83 +1132,64 @@ void Mesh::Initialize(int res_flag, ParameterInput *pin) {
     }
 
     if (res_flag == 0) {
-#pragma omp parallel for num_threads(nthreads)
-      for (int i = 0; i < nmb; ++i) {
-        MeshBlock *pmb = pmb_array[i];
-        pmb->ProblemGenerator(pin);
-      }
+      parallel_for("Problem Generator", block_range,
+                   [&](int const i) { pmb_array[i]->ProblemGenerator(pin); });
     }
 
-    int call = 0;
-    // Create send/recv MPI_Requests for all BoundaryData objects
-#pragma omp parallel for num_threads(nthreads)
+    parallel_for("Create send/recv MPI_Requests for all BoundaryData objects",
```

Comment:
Same goes for these loops. MPI calls won't work from inside kernels. 

However, I think this function is called only once, so it's not a priority to optimize.

Reply:
See previous comment.